src/components/views/rooms/MemberList.js

```diff
@@ -325,27 +348,82 @@ module.exports = React.createClass({
         });
     },
 
-    memberSort: function(userIdA, userIdB) {
-        var userA = this.memberDict[userIdA].user;
-        var userB = this.memberDict[userIdB].user;
+    memberString: function(member) {
+        if (!member) {
+            return "(null)";
+        }
+        else {
+            return "(" + member.name + ", " + member.powerLevel + ", " + member.user.lastActiveAgo + ", " + member.user.currentlyActive + ")";
+        }
+    },
 
-        var presenceMap = {
-            online: 3,
-            unavailable: 2,
-            offline: 1
-        };
+    memberSort: function(now) {
+        return (userIdA, userIdB) => {
+            // order by last active, with "active now" first.
+            // ...and then by power
+            // ...and then alphabetically.
+            // We could tiebreak instead by "last recently spoken in this room" if we wanted to.
+
+            var memberA = this.memberDict[userIdA];
+            var memberB = this.memberDict[userIdB];
+            var userA = memberA.user;
+            var userB = memberB.user;
+
+            // if (!userA || !userB) {
+            //     console.log("comparing " + memberA.name + " user=" + memberA.user + " with " + memberB.name + " user=" + memberB.user);
+            // }
+
+            if (!userA && !userB) return 0;
+            if (userA && !userB) return -1;
+            if (!userA && userB) return 1;
+
+            // console.log("comparing " + this.memberString(memberA) + " and " + this.memberString(memberB));
+
+            if (userA.currentlyActive && userB.currentlyActive) {
+                // console.log(memberA.name + " and " + memberB.name + " are both active");
+                if (memberA.powerLevel === memberB.powerLevel) {
+                    // console.log(memberA + " and " + memberB + " have same power level");
+                    if (memberA.name && memberB.name) {
+                        // console.log("comparing names: " + memberA.name + " and " + memberB.name);
+                        return memberA.name.localeCompare(memberB.name);
```

Comment:
no, it's correct: we want to return -1 when A comes before B, which is what it's doing, as per memberSort's contract.